Cargo.toml

```diff
@@ -0,0 +1,25 @@
+[package]
+name = "oxnet"
+version = "0.1.0"
+edition = "2021"
+license = "MIT OR Apache-2.0"
+description = "commonly used networking primitives with common traits implemented"
+keywords = ["networking"]
+categories = ["network-programming", "web-programming"]
+
+[features]
+default = ["serde", "schemars", "ipnetwork"]
+ipnetwork = ["dep:ipnetwork"]
+schemars = ["dep:schemars", "dep:serde_json"]
+serde = ["dep:serde"]
+std = []
```

Comment:
Oh, one other thought. Should we add an `"ip"` feature too? I imagine that we'll include things like MAC addresses here, and maybe other things, in which case one might want to only pull in support for the specific networking primitives one cares about. The `"ipnetwork"` feature would imply `"ip"` I think. Thoughts?

Reply:
Can we defer that until we have enough stuff in this crate that we want to feature-guard the categories of types? Initially I was just thinking about limited external dependencies via features.